staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1201,6 +1201,7 @@ func (c *Cacher) isStopped() bool {
 
 // Stop implements the graceful termination.
 func (c *Cacher) Stop() {
+	c.watchCache.Stop()
```

Comment:
You definitely don't want to stop it twice (given it's closing channel).

Reply:
Changed to context.CancelFunc, which can be called twice.

Reply:
Changed so Stop can be called multiple times.